src/amber/controller/helpers/asset.cr

```diff
@@ -0,0 +1,44 @@
+module Amber::Controller::Helpers
+  module Asset
+    ASSET_MANIFEST = {} of String => String
+
+    {{ run "../../run_macros/generate_asset_helpers" }}
+
+    macro asset_path(path)
+      {% if path.is_a?(StringLiteral) %}
+        {% if Amber::Controller::Helpers::Asset::ASSET_MANIFEST[path] %}
+          {{ "/dist" + Amber::Controller::Helpers::Asset::ASSET_MANIFEST[path] }}
+        {% else %}
+          {{ run "../../run_macros/missing_asset", path }}
+        {% end %}
+      {% elsif path.is_a?(StringInterpolation) %}
+        {% raise <<-ERROR
+        \n
+        The 'asset' macro doesn't work with string interpolation
+        Try this...
+          ▸ Use the 'dynamic_asset' method instead
+        ERROR %}
+      {% else %}
+        {% raise <<-ERROR
+        \n
+        The 'asset' macro requires a literal string like "my-logo.png", instead got: #{path}
+
+        Try this...
+
+          ▸ If you're using a variable, switch to a literal string
+          ▸ If you can't use a literal string, use the 'dynamic_asset' method instead
+
+        ERROR %}
```

Comment:
I'm not experienced enough with Crystal macros to have come up with this myself, and I think this is a fantastic error message. 🌮 ❗️

Reply:
Yeah, this error message looks amazing:

```
expanding macro
in lib/amber/src/amber/controller/helpers/asset.cr:12: Error executing run (exit code: 1): ../../run_macros/missing_asset "stylesheet.css"

stdout:

    "stylesheet.css" does not exist in the manifest
    Did you mean "stylesheets.css"?
```

Awesome! ✨

Reply:
Thanks to Lucky on that one :)